internal/state/connector.go

```diff
@@ -0,0 +1,65 @@
+package state
+
+import (
+	"context"
+	"time"
+
+	"github.com/ProtonMail/gluon/imap"
+)
+
+type Connector interface {
```

Comment:
Would be nice to have an explanation here about what this connector is for and how it differs from the externally-supplied top-level `connector.Connector`

Reply:
Done